membership.go

```diff
@@ -402,14 +431,24 @@ func multicastAnnounce(addr string) error {
 		logError(err)
 		return err
 	}
+	var fullLaddr string
+	if ipLen == net.IPv6len {
```

Comment:
We use the `if ipLen == net.IPv6len` idiom a lot. Perhaps we should have an `IsIPv6()` function? Just a thought.

Reply:
I had the same thought when I was typing this. I'll try to put his into a separate PR.